### ISSUE TITLE
Custom anchors to admonitions (and others) overwritten with auto-generated ones

### ISSUE THREAD

**Issue by languitar** [kai687/sphinxawesome-theme#196]
```rst
.. _foo:

.. tip::

   This is a tip!
```
With default themes this creates the following HTML code:

```html
<div class="admonition tip" id="foo">
<!-- admonition -->
</div>
```

With this theme, the div will be this version:

```html
<div class="admonition tip" id="create-a-gitlab-project-note-1">
</div>
```

This post-processing(?) breaks internal links using `:ref:`.

I suspect this also happens for heading etc.

**Comment by kai687** [comment]
It's more tricky than I thought. With admonitions, it's an easy fix. I now only only provide automatic IDs if the note doesn't already have an ID. 

With sections, etc. the mechanism seems to be a bit different. E.g. for

```
.. _foo:

This is a section
============
```

you will get something like this in HTML:

```
<div class="section" id="this-is-a-section">
   <span id="foo"></span><h1>This is a section...
```

I don't have that first span probably. So, I will have to look more into Sphinx, where the 'span' for explicit labels is generated.

**Comment by languitar** [comment]
Is this currently a result of the post-processing script? I suspect that should be easier to fix with real templating.

**Comment by kai687** [comment]
There are two parts to the answer to this question. The post-processing here is only 'adding IDs to admonitions automatically' (`admonition_ids.py`). This was not working with explicit labels, as I overwrote the IDs without checking. This is addressed by PR #197. 

Unfortunately, there isn't much in the way of templating the way Sphinx and docutils work. The templating with Jinja2 only touches the outer shell, like the header, navigation bar, and so on. Most of the reStructuredText to HTML conversion happens by docutils in Python code. For example, the default HTM5 writer class of Sphinx is [here](https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/writers/html5.py). I am inheriting from this in `html_translator.py` to overwrite quite a few of these methods to get the effects that I want at build time. 

Sphinx's HTML writer also inherits from the default `docutils` HTML writer, so I might need to look there how the explicit labels are transformed into HTML.
